common.h

```diff
@@ -0,0 +1,5 @@
+#include<iostream>
+#include<vector>
+#include<assert.h>
+#include<memory>
+using namespace std; 
```

Comment:
nit: newline at end, and space before angular braces and newline after all include statements